app/assets/javascripts/utils/api.js

```diff
@@ -1,4 +1,5 @@
 import { capitalize } from './strings';
+import NotificationActions from '../actions/notification_actions.js';
```

Comment:
> As much as possible, API.js should be strictly about querying th endpoint and returning the data

Agree. That is why I was initially working on adding this to `available_actions.jsx` where it make it sense to include a notification component.

> Following the patterns of this app would mean adding the 'NEEDS_UPDATE' case to the NotificationStore.

Ah! Makes sense. Let me make the changes and update the PR.